CodeStarterCamp_Week2/step2.swift

```diff
@@ -0,0 +1,57 @@
+func generateLottoNumbers() -> [Int] {
+    var lottoNumbers: [Int] = []
+    
+    while lottoNumbers.count < 6 {
+        let randomNumber = Int.random(in: 1...45)
+        
+        if lottoNumbers.contains(randomNumber) == false {
+            lottoNumbers.append(randomNumber)
+        }
+    }
+    
+    return lottoNumbers
+}
+
+func inputMyLottoNumbers() -> [Int] {
+    var myLottoNumbers: [Int] = []
+    
+    while myLottoNumbers.count < 6 {
+        var myInputNumber: Int = 0
+        
+        print("숫자를 입력하세요 (\(myLottoNumbers.count + 1)번째 숫자): ", terminator: "")
+        if let prompt = readLine() {
+            if let integerPrompt = Int(prompt) {
+                if integerPrompt < 1 || integerPrompt > 45 {
+                    print("잘못 입력하였습니다.")
+                    continue
+                } else {
+                    myInputNumber = integerPrompt
+                }
+            } else {
+                print("잘못 입력하였습니다.")
+                continue
+            }
+        } else {
+            print("잘못 입력하였습니다.")
+            continue
+        }
+        
+        if myLottoNumbers.contains(myInputNumber) == false {
+            myLottoNumbers.append(myInputNumber)
+        } else {
+            print("중복된 숫자입니다.")
+        }
+    }
+    
+    return myLottoNumbers
+}
+
+func compareLottoNumbers(_ myLottoNumbers: [Int], _ lottoNumbers: [Int]) -> Void {
+    let duplicatedNumbers = [Int](Set<Int>(myLottoNumbers).intersection(Set<Int>(lottoNumbers))).sorted()
```

Comment:
```suggestion
func compareLottoNumbers(_ myLottoNumbers: [Int], _ lottoNumbers: Set<Int>) -> Void {
    let duplicatedNumbers = lottoNumbers.intersection(myLottoNumbers).sorted()
```

매개변수 `lottoNumbers`의 타입을 `Set<Int>`로 바꾼다면 이정도로 간단하게 표현할 수 있겠네요

만약 그렇게 하지 않더라도 아래 코드 정도로 짧게 구현할 수 있습니다.

```swift
let duplicatedNumbers = Set(lottoNumbers).intersection(myLottoNumbers).sorted()
```

이유는 `Set`의 `intersection` 메서드가 받는 파라미터가 `Set`가 아닌 `Sequence`이기 때문인데요! [intersection(_:)](https://developer.apple.com/documentation/swift/set/intersection(_:)-6uts9) 공식문서를 통해 학습해볼까요?

또한 `intersection`을 통해 반환된 `Set`를 `Array`로 바꿔주지 않아도 되는 이유는 [sorted()](https://developer.apple.com/documentation/swift/set/sorted()) 공식문서를 봅시다!